setup_mac_zsh.sh

```diff
@@ -0,0 +1,177 @@
+echo "If you haven't downloaded the model yet, you can enter n to downloaded the model from hugging face."
+while true; do
+    read "yn?Have you already installed the model? (y/n) "
+    case $yn in
+        [Yy]* ) echo "Skipping model installation"; break;;
+        [Nn]* ) echo "Installing model"; 
+        # Prompt the user for their hugging face token and store it in a variable
+        echo "Register an account on huggingface.co and then create a token (read) on https://huggingface.co/settings/tokens"
+        echo "Also make sure to accept the disclaimer here: https://huggingface.co/CompVis/stable-diffusion-v-1-4-original"
+        read "hf_token?Please enter your hugging face token: " 
+        # Install the model
+        headertoken="Authorization: Bearer $hf_token"
+        curl -L -H "$headertoken" -o models/Stable-diffusion/sd-v1-4.ckpt https://huggingface.co/CompVis/stable-diffusion-v-1-4-original/resolve/main/sd-v1-4.ckpt 
+        break;;
+        * ) echo "Please answer yes or no.";;
+    esac
+done
+
+# Clone required repos
+git clone https://github.com/CompVis/stable-diffusion.git repositories/stable-diffusion
+ 
+git clone https://github.com/CompVis/taming-transformers.git repositories/taming-transformers
+
+git clone https://github.com/sczhou/CodeFormer.git repositories/CodeFormer
+    
+git clone https://github.com/salesforce/BLIP.git repositories/BLIP
+
+git clone https://github.com/Birch-san/k-diffusion repositories/k-diffusion
+
+# Before we continue, check if 1) the model is in place 2) the repos are cloned
+if ( [ -f "models/sd-v1-4.ckpt" ] || [ -f "models/Stable-diffusion/sd-v1-4.ckpt" ] ) && [ -d "repositories/stable-diffusion" ] && [ -d "repositories/taming-transformers" ] && [ -d "repositories/CodeFormer" ] && [ -d "repositories/BLIP" ]; then
+    echo "All files are in place. Continuing installation."
+else
+    echo "============================================="
+    echo "====================ERROR===================="
+    echo "============================================="
+    echo "The check for the models & required repositories has failed."
+    echo "Please check if the model is in place and the repos are cloned."
+    echo "You can find the model in stable-diffusion-webui/models/Stable-diffusion/sd-v1-4.ckpt"
+    echo "You can find the repos in stable-diffusion-webui/repositories/"
+    echo "============================================="
+    echo "====================ERROR===================="
+    echo "============================================="
+    exit 1
+fi
+
+# Install dependencies
+
+# In order for the requirements.txt packages to install correctly, functorch needs to be installed already.
+# functorch will not successfully build without torch, but also won't build under the nightly build we use.
+pip install torch==1.12.1
+pip install git+https://github.com/pytorch/functorch@a6e0e61242d2ed10287965c3febc426fd6abf5f9
+
```

Comment:
These lines are no longer needed and can be removed, functorch was removed as a requirement.

Reply:
Thank you! I'd already delete related code.